### PR TITLE
Implement job runner

### PR DESCRIPTION
**Pull request by jdoe7865623**

### CODE REVIEW

File: src/runner/runner.go [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+		l.Info("Loading config from %q", paths[i])
+
+		return res, nil
+	}
+
+	return nil, errors.New("config fetch failed")
+}
+
+func fetchConfigOnePath(path string) ([]byte, error) {
```
Comment (Arriven) [inline src/runner/runner.go:160]: Not required for merge, but if you get back to this PR before I get to my laptop - would be nice to rename it to something like `fetchSingleConfig`

File: src/runner/runner.go [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+	configPaths    []string
+	backupConfig   []byte
+	refreshTimeout time.Duration
+	metricsPath    string
+
+	currentRawConfig []byte // currently applied config
+
+	log *logs.Logger
+
+	stop chan interface{}
```
Comment (Arriven) [inline src/runner/runner.go:44]: Also not critical, but it's better to pass `context.Context` instead of reinventing it

**Review by Arriven — APPROVED** [review]
LGTM, will merge when I get back to laptop

File: src/runner/runner.go (hunk 1 above)
Comment (jdoe7865623) [reply src/runner/runner.go]: No probs

File: src/runner/runner.go (hunk 2 above)
Comment (jdoe7865623) [reply src/runner/runner.go:44]: I don't agree here. `context.Context` is more expensive and all we need here is `chan` to close and signal shutdown. This is a canonical concurrent operation in go.

**Pull request merged**